astropy/io/fits/fitstime.py

```diff
@@ -0,0 +1,289 @@
+    """
+    Given the time coordinate keyword information, verify that
+    each keyword has a valid value.
+    Also verify that a coordinate column of another type is not
+    mistaken to be time.
+    """
+    # FITS deprecated scales
+    global_info['scale'] = FITS_DEPRECATED_SCALES.get(global_info['scale'],
+                                                      global_info['scale'].lower())
+
+    # Verify global time scale
+    if not global_info['scale'] in Time.SCALES:
+        raise AssertionError(
+            'Global time scale (TIMESYS) must have a FITS recognized '
+            'time scale value (got {0!r}).'.format(global_info['scale']))
+
+    for idx, time_col in time_columns.items():
+        scale = time_col.get('scale', None)
+        unit = time_col.get('unit', None)
+
+        if scale is not None:
+            if scale.lower() in Time.SCALES:
+                time_col['scale'] = scale.lower()
+            elif scale in FITS_DEPRECATED_SCALES.keys():
+                time_col['scale'] = FITS_DEPRECATED_SCALES[scale]
+            elif scale == 'TIME':
+                time_col['scale'] = global_info['scale']
+            else:
+                if scale in ['GPS', 'LOCAL']:
+                    warnings.warn(
+                        'Table column {} has a FITS recognized time scale value {}. '
+                        'However, since it is not a valid astropy time scale, '
+                        'it will not be converted to astropy Time.'.format(idx, scale),
+                        AstropyUserWarning)
+                # Non-linear coordinate types have "4-3" form and are not time coordinates
+                elif TCTYP_RE_TYPE.match(scale[:5]) and TCTYP_RE_ALGO.match(scale[5:]):
+                    pass
+                else:
+                    warnings.warn(
+                        'Table column {} has a coordinate type {} which is either '
+                        'an unrecognized local time scale or a linear coordinate-'
+                        'type'.format(idx, scale),
+                        AstropyUserWarning)
+                time_columns.pop(idx)
+            continue
+
+        if (unit is not None and unit in FITS_TIME_UNIT) or 'pos' in time_col:
+            continue
+
+        # Other cases are yet to be checked and will be done when reading starts
+        # This includes checking TUNITn
+        time_columns.pop(idx)
+
+
+def _convert_time_columns(table, global_info, time_columns):
+    """
+    Convert time columns to astropy Time columns.
+
+    Parameters
+    ----------
+    table : `~astropy.table.Table`
+        The table whose time columns are to be converted.
+    global_info : dict
+        Global time reference coordinate information
+    time_columns : dict
+        Column-specific time information
+    """
+
+    # The code might fail while attempting to read FITS files not written by astropy.
+
+    # Read in Global Informational keywords as Time
+    for key in global_info:
```

Comment:
Use `for key, value in global_info.items():` to avoid all the repeats of `global_info[key]`.